pages/index.js

```diff
@@ -605,22 +578,41 @@ export const getServerSideProps = async (context) => {
 
 export default class IndexPage extends React.Component {
   async componentDidMount() {
-    const response = await Actions.health();
-    console.log("HEALTH_CHECK", response);
+    const videos = document.querySelectorAll("[data-src]");
+    const config = {
+      root: null,
+      rootMargin: "0px",
+      threshold: 0.1,
+    };
+
+    const observer = new IntersectionObserver(function (entries, self) {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          preloadVideo(entry.target);
+          self.unobserve(entry.target);
+        }
+      });
+    }, config);
+    videos.forEach((video) => {
+      observer.observe(video);
+    });
+
+    function preloadVideo(video) {
```

Comment:
This function doesn't need to be in the class right? You can declare it outside of the `IndexPage` class.